### PR TITLE
feat: restructure plugin-auth

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
restructures plugin-auth commands to the new sf command naming style guide
### What issues does this PR fix or reference?
@W-12484720@

### CODE REVIEW

File: src/commands/org/logout.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      throw messages.createError('noOrgSpecifiedWithNoPrompt');
+    }
+
+    orgAuths = this.shouldFindAllAuths(targetUsername)
+      ? await AuthInfo.listAllAuthorizations()
+      : targetUsername
+      ? await AuthInfo.listAllAuthorizations(
+          (orgAuth) => orgAuth.username === targetUsername || !!orgAuth.aliases?.includes(targetUsername)
+        )
+      : [];
```
Comment (cristiand391) [inline src/commands/org/logout.ts:136]: nit: since `orgAuths` isn't constant, could this be done with if/else conditionals? I found this a bit hard to follow (and still not sure I got it right).

File: src/commands/org/logout.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      ...orgAuths.map((orgAuth) => {
+        if (orgAuth.isScratchOrg) {
+          return 'Scratch'.length;
+        }
+        if (orgAuth.isDevHub) {
+          return 'DevHub'.length;
+        }
+        if (orgAuth.isSandbox) {
+          return 'Sandbox'.length;
+        }
```
Comment (cristiand391) [inline src/commands/org/logout.ts:88]: Should this also include org shapes?

Comment (peternhale) [reply src/commands/org/logout.ts:88]: Right now we are not attributing OrgAuthentication with orgs that are 'shapes' or 'snapshots'

File: src/commands/org/logout.ts (hunk 1 above)
Comment (peternhale) [reply src/commands/org/logout.ts:136]: I'll look into simplifying for readability, if I can figure out what it is doing ;)

**Comment by mshanemc** [comment]
qa notes

`org:list:auth` 
✅ table matches, including a mutltiple-aliases org that I have.
❓ json order is different.  It's an array so that'sfine.

`org:logout -o bad`
👎🏻  `No orgs found to log out of.` I guess? I think I'd expect, "You're not auth'd to an org with name `bad`".  There are orgs to log out of, just none with that name/alias.
👍🏻  Still an improvement over the current one which happily replies that it logged out (instead of saying that it didn't exist)
```shell
➜  plugin-auth git:(phale/restructure-auth) ✗ sfdx auth:logout -u bad
Warning: The "-u" flag has been deprecated. Use "--target-org" instead.
? Are you sure you want to log out from these org(s)?
bad

Important: You need a password to reauthorize scratch orgs. By default, scratch orgs have no password. If you still need 
your scratch orgs, run "sfdx org:generate:password" before logging out. If you don't need the scratch orgs anymore, run 
"sfdx org:delete:scratch" or "sfdx org:delete:sandbox"instead of logging out.

Log out? Yes
Successfully logged out of orgs: bad
```

same command, with --json, results in same output for this PR current sfdx.
👎🏻 I'd expect at least a warning that I may not have achieved what I meant to do. 

```
{
  "status": 0,
  "result": [],
  "warnings": []
}
```


❌  logout (with no flags except json) is gonna produce invalid json.  It needs to not do the inquirer stuff.  
➜  plugin-auth git:(phale/restructure-auth) ✗ ./bin/dev org:logout --json       
```
? Select the orgs you want to log out of: (Press <space> to select, <a> to toggle all, <i> to invert selection, and <enter> to proceed)
   Username                                    | Type    | Aliases         | Configs                             
❯◯ 1659026571028_test-z7gwz0nwf9gd@example.com |         |                 |                                     
 ◯ admin@integrationtesthubgs0.org             | DevHub  |                 |                                     
 ◯ admin@integrationtesthubna40.org            | DevHub  | na40hub,ha40hub |                                     
 ◯ aguterman@personalizedreferrals.org         |         |                 |                                     
 ◯ jduque@fx.trial                             | DevHub  | functionsHub    |                                     
 ◯ mbaizman@privacydemo.salesforce.com         |         |                 |                                     
(Move up and down to reveal more choices)
{
  "message": "Timed out after 10000 ms.",
  "name": "Error",
  "status": 1,
  "stack": "Error: Timed out after 10000 ms.\n    at Timeout._onTimeout (/Users/shane.mclaughlin/eng/plugin-auth/node_modules/@salesforce/sf-plugins-core/lib/ux/prompter.js:36:24)\n    at listOnTimeout (node:internal/timers:564:17)\n    at processTimers (node:internal/timers:507:7)",
  "exitCode": 1,
  "warnings": []
}
```

**Comment by mshanemc** [comment]
`org:logout` (with default org set)
✅ doesn't prompt me for anything, confirms that I want to delete the default org

`org:logout` (with no default org set)

✅ inquirer table.  Seems good, colors look good.
👎🏻 10s doesn't feel like enough time to find the org interactively.  This was also demo feedback last week.  

[again, that timer is rough.  This time, it's still running even after I select an org, so right before I confirm the delete, it kicks me out.  Is there a way to "reset" the timer when the command moves to the next stage, or gets some input, etc]

✅ was able to logout from 1 org using -a and then taking them all off and choosing the one I wanted.  Not going to *really* test "logout from all" because it's be too much of a pain to set it all back up.

❓ current logout had a useful message (well, with a typo, but still helpful).  Given the new multi-logout complexity, is it worth trying to keep users from abandoning scratch orgs vs. deleting them?
```
Important: You need a password to reauthorize scratch orgs. By default, scratch orgs have no password. If you still need 
your scratch orgs, run "sfdx org:generate:password" before logging out. If you don't need the scratch orgs anymore, run 
"sfdx org:delete:scratch" or "sfdx org:delete:sandbox"instead of logging out.
```

`./bin/dev org:login:access-token -r https://nosoftware-force-8563-dev-ed.scratch.my.salesforce.com -s -a tokenOrg`
✅  prompted me for token, turned it into `*`, and says it succeeded.
✅  confirmed I see it in the org:auth:list, set with correct type, and it opens fine.  org:list shows it as my local default (good)
✅  I can log out of it

**Comment by mshanemc** [comment]
❌ I have a gs0 hub that doesn't show up in auth list (new or old).  It *does* show up in the new org:list.  It doesn't exist in ~/.sfdx but does in ~/.sf/sf.json (the old single-file one).

new `org:open` won't read it.  This is probably more of a Mike question around one of the aggregators.

**Comment by mshanemc** [comment]
QA notes:

✅ jwt works fine (deprecated and correct clientId flags)
✅ org saves with alias as expected

👎🏻 jwt errors are still kinda ugly.  In this example, I sent JWT with a bad username.  I *think* this is the result of that "try several different url/audience combos" approach.
Outside this WI, I think sfdx-core should provide a better explanation than this:
```
Error (1): We encountered a JSON web token error, which is likely not an issue with Salesforce CLI. Here’s the error: Error authenticating with JWT.
Errors encountered:
user hasn't approved this consumer
user hasn't approved this consumer
user hasn't approved this consumer
user hasn't approved this consumer
user hasn't approved this consumer
```
Where maybe each error message could show what was being tried ?  

✅  sfdx-url works with correct credentials, has identical json to existing command

`sfdx auth:device -a gs0hub -r https://gs0-dev-hub... --client-id gs0...`
❌ auth:device:login prompts me for client secret.  That can't work.  Current sfdx command doesn't do that:

**Comment by peternhale** [comment]
@mshanemc regarding
sfdx auth:device -a gs0hub -r https://gs0-dev-hub... --client-id gs0...
❌ auth:device:login prompts me for client secret. That can't work. Current sfdx command doesn't do that:

There appears to be a bug in main introduced when clientid was renamed client-id See https://github.com/salesforcecli/plugin-auth/blob/main/src/commands/auth/device/login.ts#L78

I will fix main and compare results of main to my branch.

File: src/commands/org/logout.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+          )
+        );
+        // username - aliases - configs
+        const key = `${chalk.bold(
+          orgAuth.username.padEnd(maxUsernameLength)
+        )} | ${typeString} | ${aliasString} | ${chalk.yellowBright(configString)}`;
+        return { name: key, value: orgAuth, checked: all, short: `${os.EOL}${orgAuth.username}` };
+      })
+      .sort((a, b) => a.value.username.localeCompare(b.value.username));
+    const userHeader = `${'Username'.padEnd(maxUsernameLength, ' ')}`;
```
Comment (WillieRuemmele) [inline src/commands/org/logout.ts:109]: too bad we have to do all of this in the plugin

**Review by WillieRuemmele — APPROVED** [review]

File: src/commands/org/logout.ts (hunk 3 above)
Comment (mshanemc) [reply src/commands/org/logout.ts:109]: ➕
